docs/docs/rfcs/20211002_secret_management.md

```diff
@@ -1,131 +1,91 @@
+This can be solved by allowing users to register secret from Optimus CLI as a key value pair, storing them.
 
 # Technical Design
 
-To keep string literals as secret, it is a requirement Optimus keep them encrypted in database. To decrypt them, there will be two keys, one passed to Optimus server as a config value during startup and the other which clients will use to fetch registered secrets called `project key`. Project key will itself be also a secret stored like any other secret in optimus server encrypted using Optimus server key. Each secret is a key value pair where key is an alpha numeric literal and value is base64 encoded string.
+To keep string literals as secret, it is a requirement Optimus keep them encrypted in database. Optimus Server Key is used to encrypt & decrypt the secret & will ensure the secret is encrypted at rest. Each secret is a key value pair where key is an alpha numeric literal and value is base64 encoded string. 
 
-Project key will be initialised as part of project registration or project config update if it is not registered already(this is required to make project key creation backward compatible). 
+Optimus has two sets of secrets, user managed secrets & others which are needed for server operations, that will differentiate an ordinary project secret and this specific key is the prefix. Each of server managed secrets should be prefixed by `_OPTIMUS_<key name>` and will not be allowed to be used by users. Optimus should also disallow anyone using this prefix to register their secrets. The secrets can be namespaced by optimus namespace or at project level, which will help in proper authorization & access restrictions.
 
-The thing that will differentiate an ordinary project secret and this specific key is the prefix. Each of system level secret should be prefixed by `_OPTIMUS_<key name>`. Optimus should also disallow anyone using this prefix to register their secrets. So when the project gets registered and Optimus server generates the project key, it will be encrypted by Optimus key and stored with the name `_OPTIMUS_<project_name>_ROOT`. It will be used to encrypt all future secrets registered by users under this project.
-
-Reason to keep project key in db is to allow users to create new secrets without requiring them to specify the project key each time.
-
-### Project Config
-
-Optional prerequisite to registering Secrets is configuring the Kubernetes cluster details where the containers will run. This will allow Optimus to create project key in the cluster namespace so that all the containers can use this key to decrypt, once the secrets are registered, as plain text. Although this is not a hard requirement and can be done by cluster admin manually as well. This is a one time operation per registered project.
```

Comment:
You completely removed this section, I think we should add some kind of details describing how the containers will be authenticated to optimus.

Reply:
As optimus doesn't have any authentication/authorization removed this section but we can mention

Reply:
added authentication & authorization section to summarize the same.